### ISSUE TITLE
Child process extra stdio streams not implemented

### ISSUE THREAD

**Issue by Spioune** [oven-sh/bun#4670]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

Running this code in Bun and Node shows the difference in behavior.

````
const { spawn } = require('child_process');
const { createReadStream, createWriteStream } = require('fs');

const audioStream = createReadStream('./audio.m4a');
const videoStream = createReadStream('./video.mp4');
const output = createWriteStream('./output.mp4');

combineStreams(audioStream, videoStream).pipe(output)

function combineStreams(audioStream, videoStream) {
  const ffmpegProcess = spawn(
    "ffmpeg",
    [
      // '-hide_banner',
      // '-loglevel',
      // 'quiet',
      "-i",
      "pipe:3",
      "-i",
      "pipe:4",
      "-c",
      "copy",
      "-f",
      "matroska",
      "pipe:5",
    ],
    {
      stdio: [
        /* Standard: stdin, stdout, stderr */
        "ignore",
        "ignore",
        "ignore",
        /* Custom: pipe:3, pipe:4, pipe:5 */
        "pipe",
        "pipe",
        "pipe",
      ],
    }
  );
  // in nodejs ffmpegProcess.stdio = [null, null, null, <ref *1> Socket, <ref *2> Socket, <ref *3> Socket]
  // in bun ffmpegProcess.stdio = {"0": null, "1": null, "2": null}
  console.log(ffmpegProcess.stdio);
  pipeline(audioStream, ffmpegProcess.stdio[3], (err) => {});
  pipeline(videoStream, ffmpegProcess.stdio[4], (err) => {});
  return ffmpegProcess.stdio[5];
}
````

### What is the expected behavior?

I expect 3 additional streams on `stdio`

In Node, ffmpegProcess.stdio is `[null, null, null, <ref *1> Socket, <ref *2> Socket, <ref *3> Socket]`
In Bun, ffmpegProcess.stdio is `{"0": null, "1": null, "2": null}`

### What do you see instead?

In Bun, only stdin, stdout and stderr are available.

In Node, ffmpegProcess.stdio is `[null, null, null, <ref *1> Socket, <ref *2> Socket, <ref *3> Socket]`
In Bun, ffmpegProcess.stdio is `{"0": null, "1": null, "2": null}`

### Additional information

As documented here [https://nodejs.org/api/child_process.html#optionsstdio](https://nodejs.org/api/child_process.html#optionsstdio)

> ...the value of options.stdio is an array where each index corresponds to an fd in the child. The fds 0, 1, and 2 correspond to stdin, stdout, and stderr, respectively. Additional fds can be specified to create additional pipes between the parent and child.

It seems that Bun doesn't implement extra fds for child processes.

The issue is the same using `Bun.spawn()` instead of `fs.spawn()`

So right now it is not possible to pipe more than one input stream to a child process in bun.

**Comment by scsmash3r** [comment]
Bumped into this too. For me it was an issue with `curl` child process spawning...

Example:
```jsx
// [1] This works
const proc = Bun.spawn(["curl", "https://google.com"]);
const resp = await new Response(proc.stdout).text();

// [2] This works too
const proc = Bun.spawn(["curl", "--version"]);
const resp = await new Response(proc.stdout).text();

// [3] This does not, seems Bun counts this line of arguments as one option
const proc = Bun.spawn(["curl", "-ikv -X POST https://google.com"]);
const resp = await new Response(proc.stdout).text();
```

So `$ bun run index.ts` with `[3]` will generate:

```
curl: option -ikv -X POST https://google.com: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
```

**Comment by Spioune** [comment]
@scsmash3r I think this is expected behavior.
You should always separate arguments in spawn() no matter if you are using Bun or Node.js

this should work
`const proc = Bun.spawn(["curl", "-ikv", "-X", "POST", "https://google.com"]);`

**Comment by scsmash3r** [comment]
> @scsmash3r I think this is expected behavior. You should always separate arguments in spawn() no matter if you are using Bun or Node.js
> 
> this should work `const proc = Bun.spawn(["curl", "-ikv", "-X", "POST", "https://google.com"]);`

Didn't know that, thank you for the insight, that does the trick.
Probably it should be better described in docs. I'll hide my comment to avoid further confusion.

**Comment by Jarred-Sumner** [comment]
This was fixed in Bun v1.0.23 by @nektro

**Comment by Clemens-E** [comment]
I think I have the same issue:

```py
import os
from time import sleep
out = os.fdopen(3, mode='wb', buffering=0)
print('opened fd 3')
sleep(1)
out.write(b"hi from python! using fdopen 3")
out.flush()
out.close()
```

```js
import { spawn } from 'child_process';
const node = spawn('python', ['src/test.py'], {
    stdio: ['inherit', 'inherit', 'inherit', 'pipe']
});

console.log(`${node.stdio.length} fd's`)
node.stdio[3].on('data', d => console.log(d.toString()))
```

outputs different things for nodejs/bun:
```
$ node src/index.js 
4 fd's
opened fd 3
hi from python! using fdopen 3

$ bun src/index.js
4 fd's
opened fd 3
(exits)
```

I was willing to try Bun.spawn, but as it seems, it refuses to accept more than 3 values for stdio, and if you supply it (with ignore/js), I don't see any way to access extra streams.
Accessing its stdio values gives me
`[ null, null, null, 664 ]`
however not sure if I can do anything meaningful with just the file handle.

Bun v1.1.8-canary.1+dbdc37600 (Windows x64)
